verilog/formatting/tree_unwrapper_test.cc

```diff
@@ -3398,10 +3407,12 @@ const TreeUnwrapperTestData kUnwrapFunctionTestCases[] = {
         FunctionHeader(0, L(0, {"function", "foo", ";"})),
         StatementList(1,
                       FlowControl(1,  //
-                                  L(1, {"if", "(", "zz", ")"}),
-                                  NL(2, {"return", "0", ";"}),
-                                  L(1, {"else", "if", "(", "yy", ")"}),
-                                  NL(2, {"return", "1", ";"}))),
+                                  N(1,  //
+                                    L(1, {"if", "(", "zz", ")"}),
+                                    NL(2, {"return", "0", ";"})),
+                                  N(1,  //
+                                    L(1, {"else", "if", "(", "yy", ")"}),
+                                    NL(2, {"return", "1", ";"})))),
```

Comment:
suggested test case:
* two else if clauses
* else if with trailing else clause

In some of the test case comments, point out the flatness that is achieved.

Reply:
Added more tests and comments.